code/modules/mob/living/damage_procs.dm

```diff
@@ -157,6 +157,8 @@
 	if(HAS_TRAIT(src, TRAIT_NOBREATH))
 		oxyloss = 0
 		return FALSE
+	if(amount < 0 && has_status_effect(STATUS_EFFECT_NO_OXY_HEAL))
+		return FALSE
```

Comment:
just out of question, is there no current method in the code to prevent people from oxy healing already correct?

Reply:
From my code exploring and game experience, no